filebeat/input/filestream/takeover/takeover.go

```diff
@@ -154,8 +153,8 @@ func findFilestreams(log *logp.Logger, cfg *cfg.Config) (matchers filestreamMatc
 		if inputCfg.Type != "filestream" || !inputCfg.TakeOver {
 			continue
 		}
-		if inputCfg.ID == "" {
-			return matchers, errors.New("filestream `take over` mode requires a unique ID for the input")
+		if _, exists := matchers[inputCfg.ID]; exists || inputCfg.ID == "" {
+			return matchers, fmt.Errorf("filestream with ID `%s` in `take over` mode requires a unique ID. Add the `id:` key with a unique value.", inputCfg.ID)
 		}
```

Comment:
Given the `if` statement condition, this error message might not be quite true and error messages should be as precise as possible. It is better to split it into two separate checks. Also on errors, the returned `matchers` should be `nil`.

```suggestion
		if inputCfg.ID == "" {
			return nil, fmt.Errorf("filestream in `take over` mode requires a non-empty ID to be set. Add the `id:` key with a unique value.")
		}

		if _, exists := matchers[inputCfg.ID]; exists {
			return nil, fmt.Errorf("filestream in `take over` mode requires a unique ID, `%s` is not unique. Ensure the `id:` has got a unique value.", inputCfg.ID)
		}
```

I love that you also explained how to add the ID:
```
Add the `id:` key with a unique value
```
:heart_eyes: 

Feel free to change the copy form my suggestions ;)

Reply:
I think `requires to have a unique ID` also assumes that the ID must be present. I don't think we need to make it more complicated.